fili-core/src/main/java/com/yahoo/bard/webservice/data/cache/MemDataCache.java

```diff
@@ -129,8 +141,22 @@ private boolean setInSeconds(String key, T value, int expirationInSeconds) throw
         try {
             // Omitting null checking for key since it should be rare.
             // An exception will be thrown by the memcached client.
-            return client.set(key, expirationInSeconds, value) != null;
+            if (SYSTEM_CONFIG.getBooleanProperty(WAIT_FOR_FUTURE, false)) {
+                return client.set(key, expirationInSeconds, value).get();
+            } else {
+                return client.set(key, expirationInSeconds, value) != null;
```

Comment:
These two paths could use a test to guarantee config results in the correct behaviors.